core/src/main/java/org/apache/iceberg/rest/auth/OAuth2Util.java

```diff
@@ -622,7 +636,11 @@ private static AuthSession fromTokenResponse(
         String credential) {
       AuthSession session =
           new AuthSession(
-              parent.headers(), response.token(), response.issuedTokenType(), credential);
+              parent.headers(),
+              response.token(),
+              response.issuedTokenType(),
+              credential,
+              parent.scope());
```

Comment:
When this is called in `fromTokenExchange`, the scope passed there is not passed in here. I think that this method needs to have a `scope` argument. Then the `fromTokenResponse` above can pass `parent.scope()` and `fromTokenExchange` can pass the scope it was passed.

Reply:
I've simplified this to actually and removed the `scope` parameter from `fromTokenExchange`/`fromCredential` as I think we need to take the `scope` from the `parent` in those cases